### PR TITLE
Secure cookie

### PR DESCRIPTION
**Pull request by Mayur-Khante**
[1] Added secure cookie for encrypting user information stored in cookie.
[2] Unit test coverage for the secure cookie code.
[3] Unit test coverage for AWS utility code.

### CODE REVIEW

File: internal/auth.go [hunk 1]
```diff
@@ -185,19 +186,31 @@ func MakeCookie(r *http.Request, email string) *http.Cookie {
...
+			Value:    encoded,
+			Path:     "/",
+			Domain:   cookieDomain(r),
+			HttpOnly: true,
+			Secure:   !config.InsecureCookie,
+			Expires:  expires,
+		}
 	}
+
+	// if err := s.Decode(config.UserInfoCookie, cookie.Value, &value); err == nil {
```
Comment (RamuRChenchaiah) [inline internal/auth.go]: Please remove commented code

File: internal/auth_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/gorilla/securecookie"
+)
+
+func TestMakeUserCookie(t *testing.T) {
+	req, _ := http.NewRequest("GET", "http://domain.com", nil)
```
Comment (RamuRChenchaiah) [inline internal/auth_test.go]: Please move  data for tests them to setup()

File: internal/auth_test.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	}
+	value := ""
+	if err := s.Decode(userInfoCookie, got.Value, &value); err == nil && !strings.Contains(value, want.Value) {
+		return false
+	}
+	return true
+}
+
+var (
+	cookieHashKey  string = "AMC7VVW06NF6NG1BN8WGQR4GGSHYHMKN"
```
Comment (RamuRChenchaiah) [inline internal/auth_test.go]: Please move data for tests them to setup()

File: internal/aws_test.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+		fmt.Printf("bad credentials: %s", err)
+	}
+	cfg := aws.NewConfig().WithRegion(secretMgrRegion).WithCredentials(creds)
+	sess, err := session.NewSession(cfg)
+	if err != nil {
+		fmt.Println("session failed:", err)
+	}
+
+	type args struct {
+		secretMgrAccessKey string
```
Comment (RamuRChenchaiah) [inline internal/aws_test.go:30]: Please include more input validations, example: validating input length, which shall avoid buffer overflows.
(all occurrences)

File: internal/auth_test.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+	}
+
+	cookie := &http.Cookie{
+		Name:  config.UserInfoCookie,
+		Value: "test_user_info",
+	}
+
+	type args struct {
+		r        *http.Request
+		userInfo string
```
Comment (RamuRChenchaiah) [inline internal/auth_test.go:44]: Please include more input validations, example: validating input length (max length), which shall avoid buffer overflows.
(all occurrences)

File: internal/aws.go [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
+)
+
+func getAwsSession(secretMgrAccessKey, secretMgrSecretKey, secretMgrRegion string) (*session.Session, error) {
+	creds := credentials.NewStaticCredentials(secretMgrAccessKey, secretMgrSecretKey, "")
+
+	_, err := creds.Get()
+	if err != nil {
+		fmt.Printf("bad credentials: %s", err)
```
Comment (RamuRChenchaiah) [inline internal/aws.go]: Please use loggers.
(all occurrences)

**Review by RamuRChenchaiah — CHANGES_REQUESTED** [review]

File: internal/auth_test.go (hunk 5 above)
Comment (muly) [reply internal/auth_test.go:44]: thank you. length validation is taken care by `securecookie` library in the `Encode()` method. we will update the code to handle the error and propagate it up to the api layer.

**Pull request merged**